### ISSUE TITLE
Export and document catalog of footprints

### ISSUE THREAD

**Issue by nvkelso** [mojodna/marblecutter#16]
Replaces https://github.com/mojodna/marblecutter/issues/13.

**Comment by iandees** [comment]
The footprints database is a little messy. Each image that gets processed is run through `rio shapes`, which seems to generate imperfect footprints sometimes. For example, these two shapes came from the same SRTM tile:

![shapes](https://user-images.githubusercontent.com/261584/28282099-3dff9bb8-6af7-11e7-8984-320789385114.gif)

Another, different situation is that after the `rio shapes` process, we have PostGIS buffer the shape. In the case of shapes that hit the antimeridian, the buffering causes the shape to wrap around the whole world, which means it will show up in way more tiles than it should:

![image](https://user-images.githubusercontent.com/261584/28282257-d767c636-6af7-11e7-9e9b-3f692a5e2974.png)

(A broken geometry is selected and highlighted in red)

-----

The resulting footprints database will work, but it will be inefficient and might confuse people who are trying to use this to understand what sources were included in a particular tile.

**Comment by nvkelso** [comment]
@mojodna Can you comment on the buffered / broken anti-merridian (world wrapping) above, please?

Do we need to explode a multi-part, buffer, and then clip each ±180, ±90 bbox?

**Comment by nvkelso** [comment]
@iandees Can you propose documentation language and which file format(s) we're going to export the footprints in, and which s3 bucket that will live in.

**Comment by iandees** [comment]
Two format suggestions:

1. "original" format: a `pg_dump`
1. "open-er" format: something like GeoJSON or Spatialite database

**Comment by iandees** [comment]
A better example of the multiple geometries outputted by `rio shapes` is this image of a single SRTM tile ():

![image](https://user-images.githubusercontent.com/261584/28341068-139553ec-6be0-11e7-8d0f-08502e643de9.png)
*(/srtm/N43W071.tif)*

**Comment by nvkelso** [comment]
Is that one polygon feature with multiple holes?

**Comment by iandees** [comment]
No, they're all separate features. The only hole is the tiny pizza wedge near the top-center of the image: 

![image](https://user-images.githubusercontent.com/261584/28343687-861ac404-6bec-11e7-8d58-8b85e3507fc8.png)

**Comment by mojodna** [comment]
Removing `--with-nodata` seems to produce appropriate output. I'm looking into ways to correctly polygonize in order to remove the need to buffer, which should also help with the antimeridian problems.

**Comment by mojodna** [comment]
(`rio shapes` uses overviews to polygonize at lower sampling rates, somehow inscribing the actual bounds, causing groups of pixels near corners to not be included.)

**Comment by nvkelso** [comment]
So are the smaller ovals also for this one source SRTM tile?!

**Comment by iandees** [comment]
> So are the smaller ovals also for this one source SRTM tile?!

Yep! I'm wondering if `rio shapes` is building incorrect multipolygons or something. Here's the shapes overlaid on top of the SRTM image itself:

![image](https://user-images.githubusercontent.com/261584/28345542-0bd0117e-6bf9-11e7-9fa1-ddc8c454054e.png)

**Comment by mojodna** [comment]
Pretty sure that the 120 factor for multiplication is too high ;-) Can probably just buffer by `resolution` in `ingest_single_footprint.sh`.

I'm looking for a way to create accurate (pixel-encompassing) geometries that look good without being too large.

**Comment by mojodna** [comment]
Good excuse to learn me some GRASS:

```bash
r.in.gdal input=N43W071.tif output=N43W071
r.mask raster=N43W071
g.region raster=MASK
rows=$(g.region -p | grep rows | awk '{print $2}')
cols=$(g.region -p | grep cols | awk '{print $2}')
g.region rows=$[rows / 100] cols=$[cols / 100]
r.resamp.stats in=MASK out=MASK
r.to.vect input=MASK output=footprint type=area
v.out.ogr input=footprint output=footprint.json format=GeoJSON
```

Output:

http://bl.ocks.org/anonymous/raw/aaaeaa2c72ee13fc325bc060a9e41f27/

**Comment by nvkelso** [comment]
That looks great!

**Comment by nvkelso** [comment]
Updated the primary issue description with note about only exporting **enabled** footprints.

**Comment by nvkelso** [comment]
Could we tile this to say zoom 12 so we can make a Tangram map out of it?

**Comment by iandees** [comment]
Yep! I think that'd make sense.

**Comment by nvkelso** [comment]
(We talked in the meeting about using [tippecanoe](https://github.com/mapbox/tippecanoe) for this, and adding a repeatable command for under [bin](https://github.com/mojodna/marblecutter/tree/master/bin)).

**Comment by iandees** [comment]
The gzipped `pg_dump` of the footprints database is ~8.5MB! The footprints cleanup we did earlier in this ticket really helped simplify and streamline the database.

I'm currently working on testing GeoJSON and/or tippecanoe mbtiles outputs, too.

**Comment by iandees** [comment]
Found [this cool SQL on StackOverflow](https://gis.stackexchange.com/questions/124413/how-to-export-thousands-of-features-from-postgis-to-a-single-geojson-file) that spits out a valid GeoJSON file without having to use `ogr2ogr`. The resulting GeoJSON gzipped FeatureCollection is ~5.3MB.
